### PR TITLE
Merge RC branch to main for release

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR will be merged into main for the v2024.07.0 release. 

Please see https://github.com/LLNL/RAJA/pull/1695 to review the changes in the RC branch that have not yet been merged to develop.

**Approve this PR for merging**

### CODE REVIEW

**Review by adayton1 — APPROVED** [review]
LGTM

**Review by MrBurmark — APPROVED** [review]

**Comment by rhornung67** [comment]
I'm going to hold off merging this until https://github.com/LLNL/RAJA/pull/1699 is merged. Then, I'll pull that into the RC branch and do the release.

**Review by rchen20 — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -75,7 +75,7 @@ stages:
     include:
       - local: '.gitlab/custom-jobs-and-variables.yml'
       - project: 'radiuss/radiuss-shared-ci'
-        ref: 'v2024.04.0'
+        ref: 'v2024.06.0'
```
Comment (rchen20) [inline .gitlab-ci.yml:78]: Are we going with 07 or 06?

```suggestion
        ref: 'v2024.07.0'
```

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -91,20 +91,20 @@ stages:
...
-    strategy: depend
 
 include:
+  # Sets ID tokens for every job using `default:`
   - project: 'lc-templates/id_tokens'
     file: 'id_tokens.yml'
   # [Optional] checks preliminary to running the actual CI test
   - project: 'radiuss/radiuss-shared-ci'
-    ref: 'v2024.04.0'
+    ref: 'v2024.06.0'
```
Comment (rchen20) [inline .gitlab-ci.yml:107]:
```suggestion
    ref: 'v2024.07.0'
```

File: .gitlab-ci.yml (hunk 1 above)
Comment (rhornung67) [reply .gitlab-ci.yml:78]: That's the tag for the radiuss-shared-ci repo. It's independent of the RAJA release version.

File: .gitlab-ci.yml (hunk 2 above)
Comment (rhornung67) [reply .gitlab-ci.yml:107]: See comment above.

**Pull request merged**